tests/Mapper/ProgrammesDbToDomain/OptionsMapperTest.php

```diff
@@ -0,0 +1,118 @@
+    {
+        // just one set of options. no complicated inheritance
+        $options = [
+            'language' => [
+                'value' => 'cy',
+                'cascades' => true,
+            ],
+            'second_option' => [
+                'value' => null,
+                'cascades' => true,
+            ],
+        ];
+
+        $expectedOptions = [
+            'language' => 'cy',
+            'second_option' => null,
+        ];
+        $expectedEntity = new Options($expectedOptions);
+        $mapper = $this->getMapper();
+        $this->assertEquals($expectedEntity, $mapper->getDomainModel($options));
+    }
+
+    public function testGetDomainModelWithInheritance()
+    {
+        // Create three levels of options
+        // and check the ones that are allowed to cascade, do so.
+        // Once in the domain model, we don't care about the cascades property
+
+        $options = [
+            'language' => [
+                'value' => 'cy',
+                'cascades' => true,
+            ],
+            'second_option' => [
+                'value' => null,
+                'cascades' => true,
+            ],
+            'third_option' => [
+                'value' => null,
+                'cascades' => true,
+            ],
+            'fourth_option' => [
+                'value' => null,
+                'cascades' => false,
+            ],
+        ];
+        $parentOptions = [
+            'language' => [
+                'value' => 'en',
+                'cascades' => true,
+            ],
+            'second_option' => [
+                'value' => 'red',
+                'cascades' => true,
+            ],
+            'third_option' => [
+                'value' => null,
+                'cascades' => true,
+            ],
+            'fourth_option' => [
+                'value' => 'mother',
+                'cascades' => false,
+            ],
+        ];
+        $grandparentOptions = [
+            'language' => [
+                'value' => 'gd',
+                'cascades' => true,
+            ],
+            'second_option' => [
+                'value' => 'yellow',
+                'cascades' => true,
+            ],
+            'third_option' => [
+                'value' => 'monday',
+                'cascades' => true,
+            ],
+            'fourth_option' => [
+                'value' => 'grandmother',
+                'cascades' => false,
+            ],
+        ];
+
+        $expectedOptions = [
+            'language' => 'cy',
+            'second_option' => 'red',
+            'third_option' => 'monday',
+            'fourth_option' => null,
+        ];
+
+        $expectedEntity = new Options($expectedOptions);
+
+        $mapper = $this->getMapper();
+
+
+        $arrayOfAncestors = [$parentOptions, $grandparentOptions];
+
+        $this->assertEquals(
+            $expectedEntity,
+            $mapper->getDomainModel($options, ...$arrayOfAncestors)
```

Comment:
No need for the $arrayOfAncestors and using the spread operator, you could do:

`$mapper->getDomainModel($options, $parentOptions, $grandparentOptions)`